src/shared/hooks/useNSQuery.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import useUserPreferences from './useUserPreferences';
+import useNS from './useNS';
+
+import { getParamsFromURL } from '../../uniprotkb/utils/resultsUtils';
+import { getAPIQueryUrl } from '../config/apiUrls';
+import fieldsForUniProtKBCards from '../../uniprotkb/config/UniProtKBCardConfiguration';
+import { Column, nsToDefaultColumns } from '../config/columns';
+
+import { ViewMode } from '../components/results/ResultsContainer';
+import { Namespace } from '../types/namespaces';
+
+const useNSQuery = ({
+  size,
+  withFacets = true,
+}: {
+  size?: number;
+  withFacets?: boolean;
+}) => {
```

Comment:
```suggestion
} = {}) => {
```
Would avoid requiring you to pass pass an empty object where you use this hook

Reply:
👍  done